src/Workspaces/MSBuild/Core/ExternalAccess/Watch/Api/IWatchProjectFileInfoLoader.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.MSBuild.ExternalAccess.Watch.Api;
+
+internal interface IWatchProjectFileInfoLoader
+{
+    Task<LoadProjectFileInfosResult> LoadProjectFileInfosAsync(
+        string projectFilePath,
+        string languageName,
```

Comment:
Isn't language name implied by project file path?

Reply:
Sure, but there's a `ProjectFileExtensionRegistry` that gets used in MSBuild workspace. I'd prefer to leave that as an implementation detail and not expose it via the external access.